api/resourcecreator_test.go

```diff
@@ -17,55 +17,52 @@ const (
 	port            = 6900
 	resourceVersion = "12369"
 	version         = "13"
-	targetPort      = 234
 	livenessPath    = "isAlive"
 	readinessPath   = "isReady"
 	cpuRequest      = "100m"
 	cpuLimit        = "200m"
 	memoryRequest   = "200Mi"
 	memoryLimit     = "400Mi"
-	noAppConfig     = false
 )
 
 func TestService(t *testing.T) {
-
-	service := createServiceDef(targetPort, resourceVersion, appName, namespace)
+	service := createOrUpdateServiceDef(port, nil, appName, namespace)
+	service.ObjectMeta.ResourceVersion = resourceVersion
 	clientset := fake.NewSimpleClientset(service)
 
-	t.Run("Nonexistant service yields empty string and no error", func(t *testing.T) {
-		id, err := getExistingServiceId("nonexisting", namespace, clientset)
+	t.Run("Fetching nonexistant service yields nil and no error", func(t *testing.T) {
+		nonExistantService, err := getExistingService("nonexisting", namespace, clientset)
 		assert.NoError(t, err)
-		assert.Equal(t, "", id)
+		assert.Nil(t, nonExistantService)
 	})
 
-	t.Run("Existing service yields id and no error", func(t *testing.T) {
-		id, err := getExistingServiceId(appName, namespace, clientset)
+	t.Run("Fetching an existing service yields service and no error", func(t *testing.T) {
+		existingService, err := getExistingService(appName, namespace, clientset)
 		assert.NoError(t, err)
-		assert.Equal(t, resourceVersion, id)
+		assert.Equal(t, service, existingService)
 	})
 
-	t.Run("when no service exists, it's created", func(t *testing.T) {
-		service, err := createOrUpdateService(NaisDeploymentRequest{Namespace: namespace, Application: otherAppName, Version: version}, NaisAppConfig{Port: Port{TargetPort: targetPort}}, clientset)
+	t.Run("when no service exists, a new one is created", func(t *testing.T) {
+		service, err := createOrUpdateService(NaisDeploymentRequest{Namespace: namespace, Application: otherAppName, Version: version}, NaisAppConfig{Port: port}, clientset)
 
 		assert.NoError(t, err)
 		assert.Equal(t, otherAppName, service.ObjectMeta.Name)
-		assert.Equal(t, int32(targetPort), service.Spec.Ports[0].TargetPort.IntVal)
+		assert.Equal(t, int32(port), service.Spec.Ports[0].TargetPort.IntVal)
 		assert.Equal(t, map[string]string{"app": otherAppName}, service.Spec.Selector)
 	})
 	t.Run("when service exists, it's updated", func(t *testing.T) {
-		service, err := createOrUpdateService(NaisDeploymentRequest{Namespace: namespace, Application: appName, Version: version}, NaisAppConfig{Port: Port{TargetPort: targetPort}}, clientset)
-
+		newPort := 69
+		service, err := createOrUpdateService(NaisDeploymentRequest{Namespace: namespace, Application: appName, Version: version}, NaisAppConfig{Port: newPort}, clientset)
 		assert.NoError(t, err)
+
```

Comment:
Jeg syntes vi skal legge til en test på at vi ikke endrer clusterip

Reply:
da må vi sette det eksplisitt på mocken sånn at den returnerer en clusterip. Hvis vi gjør det for clusterIP bør  vi egentlig gjøre det for alle immutable felter

Reply:
Vi burde sikkert for alle. Men siden dette var en feil, og grunnen til at vi endret koden så tenkte jeg St det var et bra testcase

Reply:
lagt til